contracts/Vesting.sol

```diff
@@ -1,6 +1,131 @@
+    {
+        require(
+            initialTimestamp_ != 0,
+            "Error : 'initialTimestamp_' must be greater than 0"
+        );
+        require(vestingStartDate == 0, "error : can call only once time");
+        vestingStartDate = block.timestamp + initialTimestamp_;
+        vestingCliff = vestingStartDate + 10 minutes;
+        vestingDuration = vestingCliff + 600 minutes;
+        emit VestingStart(vestingStartDate);
+    }
+
+    function addInvestors(
+        address[] calldata investors_,
+        uint256[] calldata amount_,
+        Allocation[] calldata allocation_
+    ) external override onlyOwner {
+        require(
+            investors_.length == amount_.length &&
+                investors_.length == allocation_.length,
+            "Error : Different arrays length"
+        );
+        uint256 sumOfAmount;
+        for (uint256 i; i < investors_.length; i++) {
+            require(
+                investors_[i] > address(0) && amount_[i] > 0,
+                "Error : 'investors_' or 'amount_' , equal to 0"
+            );
+            if (allocation_[i] == Allocation.Seed) {
+                allocationList[allocation_[i]][
+                    investors_[i]
+                ] = listOfBeneficiaries[investors_[i]];
+                listOfBeneficiaries[investors_[i]].intialReward += ((amount_[
+                    i
+                ] / 10000) * 1000);
+                listOfBeneficiaries[investors_[i]].balanceBase += amount_[i];
+            }
+            if (allocation_[i] == Allocation.Private) {
+                allocationList[allocation_[i]][
+                    investors_[i]
+                ] = listOfBeneficiaries[investors_[i]];
+                listOfBeneficiaries[investors_[i]].intialReward += ((amount_[
+                    i
+                ] / 10000) * 1500);
+                listOfBeneficiaries[investors_[i]].balanceBase += amount_[i];
+            }
+            sumOfAmount += amount_[i];
+        }
+        _token.safeTransferFrom(msg.sender, address(this), sumOfAmount);
+        emit AddInvestors(investors_, amount_);
+    }
+
+    function withdrawTokens() external override {
+        require(vestingStartDate > 0, "Error : first set 'vestingStartDate'");
+        require(
+            vestingCliff < block.timestamp,
+            "Error : wait until cliff period is end"
+        );
+        require(
+            listOfBeneficiaries[msg.sender].balanceBase > 0,
+            "Error : not enougth tokens"
+        );
+        require(
+            listOfBeneficiaries[msg.sender].rewardPaid <
+                listOfBeneficiaries[msg.sender].balanceBase,
+            "Error : not enougth tokens"
+        );
+        uint256 amount = listOfBeneficiaries[msg.sender].intialReward +
+            (_calculateUnlock(100));
+        require(amount > 0, "Error : 'amount' equal to 0");
+        listOfBeneficiaries[msg.sender].intialReward = 0;
+        listOfBeneficiaries[msg.sender].rewardPaid += amount;
```

Comment:
if i not mistake in case when first user will add to Private, user claim all tokens, then user will add to Seed, this function and calculations will not work correctly